### PR TITLE
3.x: remove `test` from `testXXX` method names, create a validator

### LINKED ISSUE #6518: 3.x: remove `test` from `testXXX` method names, create a validator.
A lot of test methods were named like `testXXX` in the 0.x-1.x era and we didn't want to change those for 2.x. Now is the time to get rid of the redundant naming from all test methods.

In addition, a validator should be added that checks all test Java files and reports if it finds the pattern `void\s+test[a-zA-Z0-9]` in any file. See [this validator](https://github.com/ReactiveX/RxJava/blob/3.x/src/test/java/io/reactivex/validators/CheckLocalVariablesInTests.java) for an example of finding and checking source code files.

### PR DESCRIPTION
**Pull request by maksim-m**
Fix #6518.

- [X] Create validator
- [X] Automatically rename methods "testXXX" to "xxx" (analog to [FixLicenseHeaders](https://github.com/ReactiveX/RxJava/blob/3.x/src/test/java/io/reactivex/validators/FixLicenseHeaders.java) check)
- [x] Rename methods

### CODE REVIEW

File: src/test/java/io/reactivex/validators/TestPrefixInMethodName.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                                    } else {
+                                        lines.add(line);
+                                    }
+
+                                }
+                            } finally {
+                                in.close();
+                            }
+
+                            if (found && System.getenv("CI") == null) {
```
Comment (akarnokd) [inline src/test/java/io/reactivex/validators/TestPrefixInMethodName.java]: I'd rather not have this run unconditionally. There could be cases this introduces conflicts or other compilation errors without a good way of undoing the changes. Please comment this part out for now.

**Review by akarnokd — CHANGES_REQUESTED** [review]

File: src/test/java/io/reactivex/validators/TestPrefixInMethodName.java (hunk 1 above)
Comment (maksim-m) [reply src/test/java/io/reactivex/validators/TestPrefixInMethodName.java]: Updated

**Comment by akarnokd** [comment]
I was doing a massive set of changes to the test files myself so it would be great you rolled back the renames, kept the validator code but not fail when violations are found. I'll merge in #6526, then just the validator, then we can enable the offline/CI failure mode and fix the names in a fresh PR.

**Comment by maksim-m** [comment]
I added `@Ignore` annotation for now and removed a commit with the actual renaming.
I will open a new PR when #6526 and this one are merged.

**Review by akarnokd — APPROVED** [review]

**Review by vanniktech — APPROVED** [review]

**Pull request merged**